pkg/skuba/actions/node/upgrade/apply.go

```diff
@@ -63,6 +63,15 @@ func Apply(client clientset.Interface, target *deployments.Target) error {
 		return nil
 	}
 
+	// Refreshing cache info about target OS
+	isSUSE, err := target.IsSUSEOS()
+	// Check if target node OS is valid
+	if err != nil {
+		return err
+	}
+	if !isSUSE {
+		return err
```

Comment:
If there is no error in hasOS, and the OS is not recognized or not matching expectations (but not incorrect), isSUSE will be false, and the error will be nil. In this case, we should (at this line) return something else than nil, IMO.

However, I am not sure what _IsSUSEOS_ intended to be (as it might have been something to support openSUSE), and therefore, we might just want to ignore the error (and therefore remove the return stanza, replacing it with a warning of some sort I guess).

An experienced skuba maintainer should give an opinion here.

Reply:
Yeah, that was what I wrote before, an os.Exit and println. @chentex disagreed with this approach so I've changed the code.

Reply:
I agree with chentex on the not having os.Exist and println :) 
But I also want to make sure this return is relevant.

You could instead _produce_ an error and return it, if necessary. Or log a warning. But the code, as is, looks weird to me (return nil for error while there was an error doesn't sound right)

Reply:
Log a warning sounds fair to me.